flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraph.java

```diff
@@ -428,7 +436,11 @@ private void addEdgeInternal(Integer upStreamVertexID,
 				}
 			}
 
-			StreamEdge edge = new StreamEdge(upstreamNode, downstreamNode, typeNumber, outputNames, partitioner, outputTag);
+			if (shuffleMode == null) {
```

Comment:
Can it happen that `shuffleMode` is `null`? We should add a null check in `PartitionTransform` and mark the field `@Nonnull`.

Reply:
Yes, it can be null here. Because there might be no `PartitionTransformation` if user do not specify explicitly. And then `partitioner` and `shuffleMode` would be null here.

Reply:
Ok, makes sense